tasm-test/src/SemanticIndexTest.cc

```diff
@@ -81,6 +71,20 @@ TEST_F(SemanticIndexTestFixture, testSelectRectanglesForFrame) {
     // Ordering isn't guaranteed.
     assert(fishFrames->front() == Rectangle(1, 0, 0, 0, 0));
     assert(fishFrames->back() == Rectangle(1, 1, 0, 0, 0));
+}
 
-    std::experimental::filesystem::remove(labels);
+TEST_F(SemanticIndexTestFixture, testOpenInMemory) {
+    SemanticIndexSQLite semanticIndex(true);
+
+    std::string video("video");
+    std::string label("person");
+    for (int i = 0; i < 50; ++i) {
+      semanticIndex.addMetadata(video, label, i, 0, 0, 100, 50);
+    }
+    for (int i = 25; i < 60; ++i) {
+      semanticIndex.addMetadata(video, "cat", i, 5, 80, 20, 90);
+    }
+
+    std::shared_ptr<MetadataSelection> selectPerson(new SingleMetadataSelection(label));
+    std::shared_ptr<MetadataSelection> selectCat(new SingleMetadataSelection("cat"));
```

Comment:
I think this is a great start to the test! It would be great to also test retrieving the metadata you added, maybe by using `orderedFramesForSelection` or `rectanglesForFrame` or `rectanglesForFrames`.